src/components/DistanceRequest.js

```diff
@@ -90,16 +89,17 @@ function DistanceRequest({iou, iouType, report, transaction, mapboxAccessToken})
 
     const lastWaypointIndex = numberOfWaypoints - 1;
     const isLoadingRoute = lodashGet(transaction, 'comment.isLoading', false);
-    const hasRouteError = lodashHas(transaction, 'errorFields.route');
+    const hasRouteError = !lodashIsNil(lodashGet(transaction, 'errorFields.route'));
     const haveWaypointsChanged = !_.isEqual(previousWaypoints, waypoints);
-    const doesRouteExist = lodashHas(transaction, 'routes.route0.geometry.coordinates');
+    const doesRouteExist = TransactionUtils.doesRouteExist(transaction);
     const validatedWaypoints = TransactionUtils.getValidWaypoints(waypoints);
-    const shouldFetchRoute = (!doesRouteExist || haveWaypointsChanged) && !isLoadingRoute && _.size(validatedWaypoints) > 1;
+    const isRouteAbsentWithNoErrors = !hasRouteError && !doesRouteExist;
```

Comment:
NAB: I prefer `isRouteAbsentWithoutErrors`

Reply:
However, we also tend to prefer to name variables without negations so maybe you can refactor this a bit.